mpt_trie/src/nibbles.rs

```diff
@@ -1,6 +1,7 @@
+#![allow(clippy::assign_op_pattern)]
```

Comment:
Also just wanted to note that I think you're right in that this is the best way to get the offending macro call (`construct_uint`) to stop triggering `clippy`. From what I can tell, there isn't really a way to annotate an expanded macro "scope", so I think applying this to the entire module is the next best option. It looks like in other cases you can create an scoped block around the macro expansion with `{}`, but we can't do this here since it's this call is loose.